problem-solver/py/modules/messageProcessingModule/WeatherAgent.py

```diff
@@ -0,0 +1,110 @@
+    get_element_by_norole_relation,
+    get_system_idtf
+)
+from sc_kpm.utils.action_utils import (
+    create_action_answer,
+    finish_action_with_status,
+    get_action_arguments,
+    get_element_by_role_relation
+)
+from sc_kpm import ScKeynodes
+
+import requests
+
+
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s | %(name)s | %(message)s", datefmt="[%d-%b-%y %H:%M:%S]"
+)
+
+
+class WeatherAgent(ScAgentClassic):
+    def __init__(self):
+        super().__init__("action_show_weather")
+
+    def on_event(self, event_element: ScAddr, event_edge: ScAddr, action_element: ScAddr) -> ScResult:
+        result = self.run(action_element)
+        is_successful = result == ScResult.OK
+        finish_action_with_status(action_element, is_successful)
+        self.logger.info("WeatherAgent finished %s",
+                         "successfully" if is_successful else "unsuccessfully")
+        return result
+
+    def run(self, action_node: ScAddr) -> ScResult:
+        self.logger.info("WeatherAgent started")
+
+        message_addr = get_action_arguments(action_node, 1)[0]
+        message_type = ScKeynodes.resolve(
+            "concept_message_about_weather", None)
+
+        if check_edge(sc_types.EDGE_ACCESS_VAR_POS_PERM, message_type, message_addr):
+            try:
+                main_idtf = ScKeynodes.resolve("nrel_main_idtf", None)
+                rrel_entity = ScKeynodes.resolve("rrel_entity", None)
+                answer_phrase = ScKeynodes.resolve(
+                    "show_weather_answer_phrase", None)
+
+                # delete previous answer
+                message_answer_set = ScSet(set_node=answer_phrase)
+                message_answer_set.clear()
+
+                unknown_city_link = message_type = ScKeynodes.resolve(
+                    "unknown_city_for_weather_agent_phrase", None)
+                    
+                entity_addr = get_element_by_role_relation(
+                    src=message_addr, rrel_node=rrel_entity)
+                if not entity_addr.is_valid():
+                    create_edge(
+                        sc_types.EDGE_ACCESS_CONST_POS_PERM, answer_phrase, unknown_city_link)
+                    create_action_answer(action_node, unknown_city_link)
+                    return ScResult.OK
+
+                entity_idtf_link = get_element_by_norole_relation(
+                    src=entity_addr, nrel_node=main_idtf)
+                if not entity_idtf_link.is_valid():
+                    create_edge(
+                        sc_types.EDGE_ACCESS_CONST_POS_PERM, answer_phrase, unknown_city_link)
+                    create_action_answer(action_node, unknown_city_link)
+                    return ScResult.OK
+
+                entity_idtf = get_link_content_data(entity_idtf_link)
+                # get entity longitude and latitude
+                coordinates = requests.get(
+                    f'https://geocode.maps.co/search?city={entity_idtf}').json()[0]
```

Comment:
If connection with service will be closed or service API will be changed what was happened here? Please, add checks for this cases

Reply:
I propose to use try-catch locally, not globally (for all methods in agent body)